app/src/main/java/uk/ac/ebi/atlas/search/JsonGeneSearchController.java

```diff
@@ -130,7 +131,7 @@ public String search(@RequestParam MultiValueMap<String, String> requestParams)
         // If the query term matches a single Ensembl ID different to the query term, we return it in the response.
         // The most common case is a non-Ensembl gene identifier (e.g. Entrez, MGI, ...).
         var matchingGeneIds =
-                (geneIds.get().size() == 1 && !geneIds.get().iterator().next().equals(geneQuery.queryTerm())) ?
+                (Objects.requireNonNull(geneIds.orElse(null)).size() == 1 && !geneIds.get().iterator().next().equals(geneQuery.queryTerm())) ?
```

Comment:
I'm looking for more clarification on this as I'm confused here.`(Objects.requireNonNull(geneIds.orElse(null)).size() == 1`

Reply:
I mean in my opinion, this needs to be simplified like below:

`this.bar = Objects.requireNonNull(bar, "bar must not be null");`

or 

with static import, we can simplify further:

`this.bar = requireNonNull(bar, "bar must not be null");`

Reply:
@upendrakumbham I think you misread the code.
It is not `Objects.requireNonNull(geneIds).orElse(null)` but `Objects.requireNonNull(geneIds.orElse(null))`
Could you check it please once again?
Or we can discus it later today?
Thanks

Reply:
Ok, I found a much better way to write this code fragments.